src/main/java/core/mate/academy/service/MachineServiceImpl.java

```
@@ -1,7 +1,42 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Track;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl<T extends Machine> implements MachineService<T> {
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        if (type == Bulldozer.class) {
+            return new BuldozerProducer().get();
+        }
+        if (type == Excavator.class) {
+            return new ExcavatorProducer().get();
+        }
+        if (type == Track.class) {
+            return new TrackProducer().get();
+        }
+        return null;
```

return `null` isn't a good practice
let's throw an exception or return Collections.emptyList()

---

Done